pyomo/contrib/pynumero/algorithms/solvers/cyipopt_solver.py

```diff
@@ -0,0 +1,299 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+try:
+    import ipopt
+except ImportError:
+    raise ImportError('ipopt solver relies on cyipopt. Install cyipopt'+
+                      ' https://github.com/matthias-k/cyipopt.git')
+import numpy as np
+from pyomo.contrib.pynumero.interfaces import PyomoNLP
+import pyomo.environ as aml
+import sys
+import os
+
+
+def redirect_stdout():
```

Comment:
This is a general utility, and is very close to an implementation in `pyutilib.misc.redirect_io`.  Is there a reason that you need this implementation?

Reply:
Thanks for the suggestion @jsiirola . To be honest I was not familiar with pyutilib.misc.redirect_io. I will change cyipopt_solver.py to work with pyutilib and drop the general utilities I coded for output redirection.